noxfile.py

```diff
@@ -173,10 +174,16 @@ def pip_compile(session: nox.Session, req: str):
         arg.startswith(("-P", "--upgrade-package", "--no-upgrade")) for arg in args
     ):
         args.append("--upgrade")
+    # Use "--no-upgrade" as a signal for our noxfile to not pass --upgrade, but
+    # don't pass it on to uv. It doesn't support it.
+    with suppress(ValueError):
+        args.remove("--no-upgrade")
 
     # fmt: off
     session.run(
-        "pip-compile",
+        "uv", "pip", "compile",
+        "--universal",
+        "--quiet",
         "--output-file", f"tests/{req}.txt",
```

Comment:
Should we also specify a Python version here?

Reply:
Do you mean like if you were to get the python version from the session?

```
python_version = f"{session.python}"

# Then add a constraint in the args
session.run(
        "uv", "pip", "compile",
        "--universal",
        "--quiet",
        f"--python-version={python_version}",
```

Wouldn't that be an unnecessary duplication `@nox.session(name="pip-compile", python=["3.11"])`? Maybe I'm missing something...